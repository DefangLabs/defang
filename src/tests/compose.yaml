name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
