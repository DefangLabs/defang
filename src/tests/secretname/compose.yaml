services:
  app:
    restart: unless-stopped
    build:
      context: .
    secrets:
      - dummy
secrets:
  dummy:
    external: true
    name: dummyx
