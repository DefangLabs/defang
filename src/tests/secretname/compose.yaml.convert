[
  {
    "name": "app",
    "platform": 2,
    "internal": true,
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile"
    },
    "secrets": [
      {
        "source": "dummy"
      }
    ],
    "networks": 1
  }
]