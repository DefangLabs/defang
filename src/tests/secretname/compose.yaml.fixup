[
  {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile"
    },
    "command": null,
    "entrypoint": null,
    "environment": {
      "dummy": null
    },
    "networks": {
      "default": null
    },
    "restart": "unless-stopped"
  }
]