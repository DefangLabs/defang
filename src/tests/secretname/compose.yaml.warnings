level=warning msg="service \"app\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"