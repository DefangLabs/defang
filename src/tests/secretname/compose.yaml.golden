name: secretname
services:
  app:
    build:
      context: ../../tests/secretname
      dockerfile: Dockerfile
    networks:
      default: null
    restart: unless-stopped
    secrets:
      - source: dummy
        target: /run/secrets/dummy
networks:
  default:
    name: secretname_default
secrets:
  dummy:
    name: dummyx
    external: true
